ast/Helpers.h

```diff
@@ -546,16 +546,26 @@ class MK {
         return send->fun == core::Names::new_() && send->recv.isSelfReference();
     }
 
+    /*
+     * Is this an expression that refers to `::T`?
+     *
+     * This helper works on a purely syntactic level. For instance, this function determines if an expression is `T`,
+     * either with no scope or with the root scope (i.e. `::T`). This might not actually refer to the `T` that we define
+     * for users, but we don't know that information at the AST level.
+     */
     static bool isT(const ast::ExpressionPtr &expr) {
         bool result = false;
 
         typecase(
             expr,
             [&](const ast::UnresolvedConstantLit &t) {
+                // When the `T` was written by the user, we get an UnresolvedConstantLit.
                 result = t.cnst == core::Names::Constants::T() && ast::MK::isRootScope(t.scope);
             },
-            [&](const ast::ConstantLit &c) { result = c.symbol() == core::Symbols::T(); },
-            [&](const ast::ExpressionPtr &e) { result = false; });
```

Comment:
Not sure why we would need this branch when we already set the result at `false`?

Reply:
```

[2025-02-25T14:57:28Z] ERROR: /app/payload/binary/BUILD:71:8: Executing genrule //payload/binary:generate-state-payload-raw failed: (Exit 10): bash failed: error executing command (from target //payload/binary:generate-state-payload-raw) /bin/bash -c ... (remaining 1 argument skipped)
--
  | [2025-02-25T14:57:28Z] Exception::raise() msg="not handled typecase case: 31"
  | [2025-02-25T14:57:28Z]
  | [2025-02-25T14:57:28Z] Backtrace:
  | [2025-02-25T14:57:28Z]   #3 0x562d8696b562 sorbet::rewriter::TypeAssertion::run()
  | [2025-02-25T14:57:28Z]   #4 0x562d8695fd04 sorbet::rewriter::Rewriterer::postTransformSend()

```

is why. 😅

Reply:
Oh I didn't realize it was a runtime issue, I ran a bunch of tests locally that didn't go through this code path 😬